addons/mrp/models/mrp_production.py

```diff
@@ -643,11 +665,11 @@ def _compute_unreserve_visible(self):
             order.unreserve_visible = not any_quantity_done and already_reserved
             order.reserve_visible = order.state in ('confirmed', 'progress', 'to_close') and any(move.product_uom_qty and move.state in ['confirmed', 'partially_available'] for move in order.move_raw_ids)
 
-    @api.depends('workorder_ids.state', 'move_finished_ids', 'move_finished_ids.quantity')
+    @api.depends('workorder_ids.state', 'move_finished_ids', 'move_finished_ids.quantity', 'product_uom_id')
     def _get_produced_qty(self):
         for production in self:
-            done_moves = production.move_finished_ids.filtered(lambda x: x.state != 'cancel' and x.product_id.id == production.product_id.id)
-            qty_produced = sum(done_moves.filtered(lambda m: m.picked).mapped('quantity'))
+            done_moves = production.move_finished_ids.filtered(lambda x: x.state != 'cancel' and x.product_id.id == production.product_id.id and x.picked)
+            qty_produced = sum(m.product_uom._compute_quantity(m.quantity, production.product_uom_id) for m in done_moves)
```

Comment:
Why that change? The `move_finished_ids` for the end-product of a MO is already in the uom of the MO itself, so we wouldn't need any conversion here?

Reply:
![image](https://github.com/user-attachments/assets/24b2ca45-187b-4eaa-a2d1-1c793c72853f)
It was part of PO's review that the transfer resulting from MOs should be in the base unit of the product. So, I had to force the unit of the product on the finished move: https://github.com/odoo/odoo/pull/184131/files#diff-6f4b1d411b3160a63b8260bf87166f0f20452b9b9fe3f04d28c3734302509f05R1171. I'm not sure though if this breaks other flows.

Reply:
I think that's a very risky move, as it's pretty much always assumed that there's a mirror between the MO uom/product_qty/... and the finished_move of the end product itself, as it's the only was you can change it (since `move_finished_ids` are only shown for by-products otherwise).